dataland-frontend/tests/e2e/fixtures/sme/SmeDataFixtures.ts

```diff
@@ -1,103 +1,112 @@
       },
       companyFinancials: {
-        revenueInEur: valueOrUndefined(randomInt(100000000), undefinedProbability),
-        operatingCostInEur: valueOrUndefined(randomInt(80000000), undefinedProbability),
-        capitalAssetsInEur: valueOrUndefined(randomInt(70000000), undefinedProbability),
+        revenueInEur: dataGenerator.randomInt(100000000),
+        operatingCostInEur: dataGenerator.randomInt(80000000),
+        capitalAssetsInEur: dataGenerator.randomInt(70000000),
       },
     },
     production: {
       sites: {
-        listOfProductionSites: valueOrUndefined(generateArray(generateProductionSite), undefinedProbability),
+        listOfProductionSites: dataGenerator.generateProductionSite(),
       },
       products: {
-        listOfProducts: valueOrUndefined(generateArray(generateProduct), undefinedProbability),
+        listOfProducts: dataGenerator.generateProduct(),
       },
     },
     power: {
       investments: {
-        percentageOfInvestmentsInEnhancingEnergyEfficiency: valueOrUndefined(
-          getRandomPercentageRangeInvestmentEnergyEfficiency(),
+        percentageOfInvestmentsInEnhancingEnergyEfficiency: dataGenerator.valueOrUndefined(
+          generatePercentageRangeInvestmentEnergyEfficiency(),
         ),
       },
       consumption: {
-        powerConsumptionInMwh: valueOrUndefined(randomInt(2000), undefinedProbability),
-        powerFromRenewableSources: valueOrUndefined(randomYesNo(), undefinedProbability),
-        energyConsumptionHeatingAndHotWater: valueOrUndefined(randomInt(1000), undefinedProbability),
-        primaryEnergySourceForHeatingAndHotWater: valueOrUndefined(getRandomHeatSource()),
-        energyConsumptionCoveredByOwnRenewablePowerGeneration: valueOrUndefined(
-          getRandomPercentageRangeEnergyConsumption(),
+        powerConsumptionInMwh: dataGenerator.valueOrUndefined(generateInt(2000)),
+        powerFromRenewableSources: dataGenerator.randomYesNo(),
+        energyConsumptionHeatingAndHotWater: dataGenerator.randomInt(1000),
+        primaryEnergySourceForHeatingAndHotWater: dataGenerator.valueOrUndefined(generateHeatSource()),
+        energyConsumptionCoveredByOwnRenewablePowerGeneration: dataGenerator.valueOrUndefined(
+          generatePercentageRangeEnergyConsumption(),
         ),
       },
     },
     insurances: {
       naturalHazards: {
-        insuranceAgainstNaturalHazards: valueOrUndefined(randomYesNo(), undefinedProbability),
-        amountCoveredByInsuranceAgainstNaturalHazards: valueOrUndefined(randomInt(50000000), undefinedProbability),
-        naturalHazardsCovered: valueOrUndefined(getRandomSlectionOfNaturalHazards(), undefinedProbability),
+        insuranceAgainstNaturalHazards: dataGenerator.randomYesNo(),
+        amountCoveredByInsuranceAgainstNaturalHazards: dataGenerator.randomInt(50000000),
+        naturalHazardsCovered: dataGenerator.valueOrUndefined(generateSelectionOfNaturalHazards()),
```

Comment:
Potential usage of the "randomArray()" feature of the data generator

Reply:
refactored